### ISSUE TITLE
Add support for Stripe's "new" checkout flow

### ISSUE THREAD

**Issue by allendav** [woocommerce/woocommerce-gateway-stripe#799]
https://stripe.com/docs/payments/checkout

Instead of completing checkout on the merchant's site, this new checkout flow is hosted by Stripe. Unlike the modal checkout we currently offer, this flow includes SCA support. It looks like this:

<img width="1239" alt="screen shot 2019-02-21 at 11 39 59 am" src="https://user-images.githubusercontent.com/1595739/53196730-8f48ee00-35cd-11e9-9b09-e501872c11ad.png">

-------------------

- [ ] Issue assigned to next milestone.
- [ ] Issue assigned a priority (will be assessed by maintainers).

**Comment by BGogurt** [comment]
It would be great to be able to use the new Checkout in order to use Stripe's Chargeback Protection as well.

**Comment by majekagriffin** [comment]
Is there an estimated date for the upgrade of the stripe plugin to supports Stripe's chargeback protection?

**Comment by madeincosmos** [comment]
Also reported in 2209434-zen

**Comment by iMazed** [comment]
Stripe is recommended folks to implement code to add Checkout compatibility: 2242306-zen. To prevent confusion for customers (and extra work for Happiness), perhaps this can be looked at?

**Comment by iMazed** [comment]
Setting to high as SCA comes into play at September 14th.

**Comment by johnlmccurdy** [comment]
Stripe is now really ramping up their communications re: the SCA regulations. What is the eta on this feature as it would be preferable to using existing woocommerce / theme templates also???

**Comment by pablogiralt** [comment]
Hi, any updates?

**Comment by majekagriffin** [comment]
is there any further updates yet?  Rough ETA would be very much appreciated

**Comment by PeteAH** [comment]
Any updates? This is great functionality and it would be great to have it in time for the Xmas season

**Comment by rwaynick** [comment]
Another vote for new checkout support to get the chargeback protection feature in 16190343-hc

**Comment by ozrobotics** [comment]
Please do add Stripe's new Chargeback Protection feature to your Stripe Plugin ASAP! With the growing online sales there has been a huge increasing number in fraudulent purchases especially nowadays! We have described this very important matter to WooCommerce team many times for specific reasons that are crucial to all website owners who use WooCommerce with Stripe.

**Comment by jirikrepl** [comment]
Can we somehow easily add "G Pay" button with the current Stripe Woocomerce plugin? Or having Stripe Checkout Woocomerce plugin would be the only easy way?
<img width="421" alt="Screen Shot 2019-12-03 at 13 59 02" src="https://user-images.githubusercontent.com/2588479/70053306-6d0ce380-15d5-11ea-8a10-35fa734715f4.png">

**Comment by dougaitken** [comment]
Hi @jirikrepl - those buttons are Payment Request buttons and not related to the Checkout feature.

We are tracking this button styling in #1071

**Comment by digisavvy** [comment]
I would like to see stripe checkout support added as well.

**Comment by jrick1229** [comment]
Reporting in 18675359-hc

**Comment by lesloups-uk** [comment]
Any news ?

**Comment by lesloups-uk** [comment]
Would you have a rough ETA now per chance?

**Comment by ozrobotics** [comment]
Yes, the Stripe's new chargeback protection is extremely important for all ecommerce websites: https://stripe.com/gb/radar/chargeback-protection - We have been talking to Stripe and they have confirmed that WooCommerce Stripe plugin needs some modification in order to work with their new checkout for https://stripe.com/gb/radar/chargeback-protection. Thank you.

**Comment by lesloups-uk** [comment]
> Yes, the Stripe's new chargeback protection is extremely important for all ecommerce websites: https://stripe.com/gb/radar/chargeback-protection - We have been talking to Stripe and they have confirmed that WooCommerce Stripe plugin needs some modification in order to work with their new checkout for https://stripe.com/gb/radar/chargeback-protection. Thank you.

Thank you for your quick answer, may we have an idea for when this could be made available? thank you for your hard work !

**Comment by lesloups-uk** [comment]
?

**Comment by lesloups-uk** [comment]
any news ?

**Comment by dougaitken** [comment]
Hi @lesloups-uk - when there is news, there will be an update here. If there is no update, there is no news to share. Please don't continue pinging on this issue. 

Thanks,

**Comment by PeteAH** [comment]
Any updates? Been over a year now?

**Comment by chwba** [comment]
Any news?

**Comment by jwiebels** [comment]
Would be interested in this as well.

**Comment by lesloups-uk** [comment]
an approximative ETA would indeed mean a lot, the absence of the new stripe checkout means literally 1K£ of profit lost for me every month..

**Comment by lesloups-uk** [comment]
Would anyone know a quick and not too complicated way to switch to the stripe new checkout system (and code) so we could use the chargeback protection in the meantime?

I'm reading about it but it seems like a real hassle switching to Stripe code and maintaining compatibility with the Woocommerce store and products

**Comment by batuhanguldemir** [comment]
![image](https://user-images.githubusercontent.com/50812784/80287320-94597e00-8739-11ea-8efa-ea93c8f411a5.png)

I want to translate these fields into Turkish. How can I do it.
Card number ,expiry date ,card code (CVC)

**Comment by dougaitken** [comment]
@batuhanguldemir This Issue is around a different discussion. 

This may help guide you in the right direction - https://github.com/woocommerce/woocommerce-gateway-stripe/issues/521

If that doesn't help, please post in the Stripe forums - https://wordpress.org/support/plugin/woocommerce-gateway-stripe/ - or reach out to the WooCommerce.com support team - https://woocommerce.com/my-account/create-a-ticket/

Thanks

**Comment by Pairfum-London** [comment]
We would also very much welcome the implementation of the "New Stripe Checkout" within the plugin in order for us to use Stripe's Chargeback Protection.

**Comment by chwba** [comment]
For the meantime: https://wordpress.org/plugins/payment-gateway-stripe-and-woocommerce-integration/

This plugin can do it, even in the free version.

**Comment by jacoswan** [comment]
Also in 3092176-zen

**Comment by jacoswan** [comment]
22518368-hc

**Comment by dougaitken** [comment]
Another related from the forums - https://wordpress.org/support/topic/chargeback-protection-stripe-checkout-sca-compliance

**Comment by Beats24-7** [comment]
Hello,
any update on this matter? Has it been implemented WooCommerce yet or do we still need to use the external plugin linked above? Thanks

**Comment by damiannep** [comment]
Requested in 25803058-hc

**Comment by devsigner-xyz** [comment]
This would be an awesome feature.

**Comment by graemeblake** [comment]
Any updates or roadmap? Would *love* to be able to use Stripe Checkout directly

**Comment by Beats24-7** [comment]
Hello,
any update on this? It has really been a while. Thanks!

**Comment by 0chak** [comment]
Can someone explain why this feature was not implemented yet? 
Right now the only way to use stripe is by displaying a credit card form inside the payment method selector, you can't even use the old stripe popup/modal anymore, this is a real issue for me because my theme messes up that form completely, and anyways I don't want to deal with the customer credit card info and **I don't want the customer to think that he's giving his credit card info to me.**
It would be very nice to be finally able to use that external checkout page like I do with paypal checkout.

**Comment by igotdes** [comment]
Also requested in #30144948-hc/#4069215-zen.

**Comment by bstonedev** [comment]
I also support adding Stripe Checkout to plugin.

I note that this plugin support forum thread ([wordpress.org/support/topic/stripe-checkout-chargeback-protection-feature/ ](wordpress.org/support/topic/stripe-checkout-chargeback-protection-feature/)) states that Chargeback Protection is possible with existing plugin. My understanding is that this is incorrect.

**Comment by darcie** [comment]
Requested in 4211218-zen.

**Comment by daniyalahmadk** [comment]
Requested in 4561326-zen.

**Comment by aydenbottos** [comment]
It is now 2022 and yet no progress? Anyone trying to sell to Europe can’t use this plugin AT ALL. It has been 3 years since this was first proposed. Come on, WooCommerce.

**Comment by lkraav** [comment]
> It is now 2022 and yet no progress? Anyone trying to sell to Europe can’t use this plugin AT ALL. It has been 3 years since this was first proposed. Come on, WooCommerce.

Wdym "sell to Europe, can't use this plugin at all".

We're selling worldwide just fine. What am I missing?

**Comment by aydenbottos** [comment]
The plugin doesn't support SCA, which is necessary for anyone selling to
consumers in the European Economic Area.

**Comment by dougaitken** [comment]
@aydenbottos - This Stripe integration fully support SCA/PSD2 and has since this was launched - https://woocommerce.com/document/stripe/#sca-psd2 / https://woocommerce.com/posts/introducing-strong-customer-authentication-sca/

**Comment by aydenbottos** [comment]
Oh makes sense, sorry about that I believed that the new checkout flow was
required. Still, it is something that needs to be done after 3 YEARS.

**Comment by aydenbottos** [comment]
Any updates on this? It's June 2022, nearly 3 AND A HALF YEARS since it was first asked for.

**Comment by maltfield** [comment]
Any update on this? I want to add fiat payments to my woocommerce site without having to poke a hole in my firewall to permit my web server to be able to communicate to api.stripe.com.

My understanding is that Stripe Checkout is a "client-only" solution, which -- among other benefits -- is more secure for the customer (the whole payment transaction is done on stripe's servers, not the merchant's servers).

See this thread for more context:

 * https://wordpress.org/support/topic/is-server-api-stripe-com-really-necessary-firewall/

Related: lots of folks ITT complaining about chargebacks. If this is your concern, consider accepting cryptocurrencies and offering a discount to your customers. Fees are much less than Stripe and chargebacks don't exist in the crypto ecosystem. Personally, I've had great success with this plugin. Integration was way simpler than Stripe, and it's trivial to accept over 2,000 cryptocurrenices.

 * https://wordpress.org/plugins/coinpayments-payment-gateway-for-woocommerce/

**Comment by EjayhanFernandes** [comment]
Requested here: 5464446-zen

**Comment by thisissandip** [comment]
5478700-zen

**Comment by nanoanno** [comment]
Additional request: 5486591-zen

**Comment by donwattz1459** [comment]
Would love this integration. Seems like 3 years since the topic opened.

**Comment by gomescx** [comment]
Bump up on this request. 
My personal reason is to allow customers to use AfterPay and BECS, which are available on Stripe Checkout but not available as an additional payment type on <Stripe for Woocommerce>

**Comment by roryWolf** [comment]
Another request here: 5899790-zd

**Comment by imodouglas** [comment]
6082341-zen

**Comment by OzanB05** [comment]
Hello, why is the apple payment not working for bundles. We have in our online shop a Bundle of 10 Products. But then the apple payment is not working. Online, if you have a single product in your cart, the apple payment works. I need help, please.

**Comment by Swapot** [comment]
I'm guessing this is not happening for Woocommerce Stripe since this is many years old now?  Would be nice - much better flow. Looks very similar to Shopify checkout.

<img width="1433" alt="Screen Shot 2024-01-09 at 10 38 05 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/152375535/e008c50b-7b12-4a63-8e94-191578d0dfec">

**Comment by aydenbottos** [comment]
It's ridiculous that after 5 years, we still haven't seen such a great feature implemented. Can anyone comment to the status of this?

**Comment by mountainash** [comment]
@aydenbottos it's not "ridiculous" at all - it's open source! If you want something done badly enough you can do it yourself.

**Comment by aydenbottos** [comment]
@mountainash I would normally contribute here, but I’m not a front-end developer and work with C/Rust/Python etc instead of PHP and JS in this codebase. I see that you’ve created a few issues for feature requests - it’s not a foreign concept mate and after 5 years, it would be nice to have this feature.

**Comment by mountainash** [comment]
You've been complaining on this thread for 3 years - it's not adding any value. It's also plenty of time to have up-skilled, become full-stack and give back to something you seem to care about. 💭

**Comment by aydenbottos** [comment]
Thanks for your concern mate, but my work in backend SWE serves me just fine. I only joined this thread to complain as I have clients with frontend teams that aren’t able to use this feature.

**Comment by mountainash** [comment]
And complain you have... multiple times - how's that working for you (multiple years later)?

Rather than add unnecessary angst on the Maintainers of this free software - maybe you could put some energy into showing the Frontenders you support how easy it is to contribute (good and bad - but maybe more good) back too the projects they make money on? And in the case of this project; they really are collecting money!

Also... I'm not your "mate".

**Comment by Pairfum-London** [comment]
This comment 'war' is at risk of becoming a bit petty.

We would be happy to contribute financially towards getting this functionality developed.

Who could develop it?
Who else is prepared to chip in financially?

Perhaps this might be a way forward out of the deadlock.

**Comment by aydenbottos** [comment]
I would also be happy to contribute financially to fund this development. It would be a great addition to the plugin.

Mountainash, I don’t work with these frontend teams and doubt they have the spare time to contribute here. I apologize for calling you mate - I saw you’re an Aussie as well and it’s just a shame it had to come to this.

**Comment by maltfield** [comment]
Hey bot, looks like lots of people still want this.

**Comment by aydenbottos** [comment]
6 years on, yes, we are still wanting this to be added

**Comment by Beats24-7** [comment]
+1

**Comment by conraid** [comment]
It would be useful to have it, the important thing is that you don't do something like the paypal plugin that broke compatibility with old subscriptions in subscriptions

**Comment by Swapot** [comment]
Yes, I would use the Stripe's "new" checkout flow if it were added to the Wordpress plugin. This would give people like me, who aren't developers, the ability to have a more user-friendly checkout like Shopify has.

**Comment by brucevalle** [comment]
+1

**Comment by rakesh535** [comment]
+1

**Comment by graemeblake** [comment]
Still interested for sure

**Comment by bgoewert** [comment]
Not stale. Also not completed?

**Comment by 0chak** [comment]
I know that paypal does not like merchants to mix stripe and paypal, (I think it's even written in their ToS) so because woocommerce went all in with paypal as a default payment provider, even though integrating stripe is super easy it would create business conflicts

**Comment by wjrosa** [comment]
Closing as obsolete. This request predates the plugin's move to the Payment Intents API and the Payment Element (UPE): the modal Stripe Checkout it compares against was removed years ago, and SCA support, the main motivation here, has long been covered by the Intents-based checkout. The plugin's direction is embedding Stripe Elements in the WooCommerce checkout rather than redirecting to a Stripe-hosted page.

> *Drafted with AI; reviewed by me.*

**Comment by daledupreez** [comment]
I am going to re-open this as I think the decision here is more nuanced. While the original request was focused on SCA support, many merchants have requested support for Stripe's hosted checkout for other reasons, including integration with other Stripe services and features. I think we should take a more considered and broader look at whether this is something we want to support.

**Comment by dashkevych** [comment]
Growing share of Stripe's app ecosystem attaches to hosted Checkout Sessions rather than Elements. Surcharging, fee handling and compliance-disclosure apps all operate on hosted sessions. It would be great if the plugin could offer support for Stripe's hosted checkout 🤞
